services/galley/test/integration/API/Util.hs

```diff
@@ -2792,10 +2807,60 @@ wsAssertBackendRemoveProposal fromUser convId kpref n = do
       case rmValue rp of
         RemoveProposal kpRefRemove ->
           kpRefRemove @?= kpref
-        otherProp -> error ("Exepected RemoveProposal but got " <> show otherProp)
-    otherPayload -> error ("Exepected ProposalMessage but got " <> show otherPayload)
+        otherProp -> assertFailure $ "Expected RemoveProposal but got " <> show otherProp
+    otherPayload -> assertFailure $ "Expected ProposalMessage but got " <> show otherPayload
+  pure bs
+  where
+    getMLSMessageData :: Conv.EventData -> ByteString
+    getMLSMessageData (EdMLSMessage bs) = bs
+    getMLSMessageData d = error ("Excepected EdMLSMessage, but got " <> show d)
+
+wsAssertAddProposal ::
+  HasCallStack =>
+  Qualified UserId ->
+  Qualified ConvId ->
+  Notification ->
+  IO ByteString
+wsAssertAddProposal fromUser convId n = do
```

Comment:
Given this one is MLS-specific, maybe a better place for it would be the `MLS.Util` module.

Reply:
I think it's fine to have all these event assertions there.